src/pipeline_migration/actions/modify/generic.py

```diff
@@ -0,0 +1,258 @@
+import argparse
+import copy
+import logging
+from pathlib import Path
+from typing import Any, Final
+
+from ruamel.yaml.comments import CommentedSeq, CommentedMap
+
+from pipeline_migration.yamleditor import EditYAMLEntry, YAMLPath
+from pipeline_migration.types import FilePath
+from pipeline_migration.utils import YAMLStyle, create_yaml_obj
+from pipeline_migration.pipeline import PipelineFileOperation, iterate_files_or_dirs
+
+
+logger = logging.getLogger("modify.generic")
+
+
+SUBCMD_DESCRIPTION: Final = """\
+The following are several examples with a raw yaml modification:
+
+* Modify an yaml item within relative .tekton/ directory.
+
+    cd /path/to/repo
+    pmt modify generic insert '["path", "to", "yaml", "item"]' '{"new": "item"}'
+
+* Modify an yaml item in multiple pipelines in several repositories:
+
+    pmt modify \\
+        -f /path/to/repo1/.tekton/pr.yaml -f /path/to/repo2/.tekton/push.yaml \\
+        generic replace \\
+        '["path", "to", "yaml", "item", 3]' '{"replaced": "new"}'
+
+* Remove an task using yq's `path` function:
+
+   pmt modify \\
+        -f .tekton/pr.yaml \\
+        generic remove \\
+        "$(yq '.spec.pipelineSpec.tasks[] | select(.name == "prefetch-dependencies") | \\
+            path' .tekton/pr.yaml)"
+
+WARNING: generic subcommand should be used as the last resort subcommand, it doesn't do any
+semantic validation for Konflux tasks.
+Use resource specific subcommands if they are available instead to have proper validation.
+"""
+
+
+def register_cli(subparser) -> None:
+    mod_generic_parser = subparser.add_parser(
+        "generic",
+        help=(
+            "Generic modification of YAML file (specific resource subcommands should "
+            "be preferred)"
+        ),
+        description=SUBCMD_DESCRIPTION,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+
+    subparser_mod = mod_generic_parser.add_subparsers(
+        title="subcommands to generic modifications", required=True
+    )
+
+    # insert
+    subparser_insert = subparser_mod.add_parser(
+        "insert",
+        help="Inserts item into YAML path",
+    )
+    subparser_insert.add_argument(
+        "yaml_path",
+        help=(
+            "YAML path (in YAML format). Must point to a sequence or a map item. "
+            "It's the same path as returned by yq's path function (list of indexes)"
+        ),
+        metavar="YAML-PATH",
+    )
```

Comment:
We have to describe the supported YAML path syntax in detail. `SUBCMD_DESCRIPTION` is a good place.

yq's `path` function outputs in this form:

```
- spec
- steps
- 1
- script
```

It is required to convert it to the decoded JSON list string in order to pass to `generic` subcommand.

It would be good if `generic` subcommand accepts a string representation of a YAML path, like `spec.tasks.1.params`. It requires less effort on editing the decoded JSON list string, whatever by hand or a command line converting the output from `path` function.

Reply:
What do you mean?